validator/sawtooth_validator/state/merkle.py

```diff
@@ -31,6 +31,28 @@
 TOKEN_SIZE = 2
 
 
+def _decode(encoded):
+    return cbor.loads(encoded)
+
+
+def _encode(value):
+    return cbor.dumps(value, sort_keys=True)
+
+
+def _hash(stuff):
+    return hashlib.sha512(stuff).hexdigest()[:64]
+
+
+def _encode_and_hash(value):
+    packed = _encode(value)
+    return _hash(packed), packed
+
+
+NODE_PROTO_PACKED = _encode(NODE_PROTO)
```

Comment:
Note that because these are constants, there's no need to calculate these in the code at all -- the calculations could be done in the REPL and then the literal results could be inserted into the code. That would be slightly more efficient, but at a severe cost to maintainability.

Instead, calculate them once at the start. A separate `_encode` function is used here because the calculations are done prior to the initialization of any particular `MerkleDatabase`, so `MerkleDatabase._encode` can't be used. I don't think `MerkleDatabase.hash` can be used either because of the way the statements are ordered in the file.